aiocache/backends/memcached.py

```diff
@@ -147,8 +148,8 @@ def __init__(self, serializer=None, **kwargs):
     def parse_uri_path(cls, path):
         return {}
 
-    def _build_key(self, key, namespace=None):
-        ns_key = super()._build_key(key, namespace=namespace).replace(" ", "_")
+    def build_key(self, key: str, namespace: Optional[str] = None) -> str:
```

Comment:
Should probably try to diff the mypy results to see if there are new errors. master currently has 20 errors:
https://github.com/aio-libs/aiocache/actions/runs/4235473010/jobs/7359129847

This must be an error, because the function returns `bytes`.

Maybe we need a `TypeVar` to declare the type here (which can be defined in each subclass).

Reply:
Or 12 errors after installing all missing libraries:
https://github.com/aio-libs/aiocache/actions/runs/4235831591/jobs/7359931126

Reply:
To use a `TypeVar` in the return type, we'll also need either:
1. The same `TypeVar` to annotate on the function parameters.  We could add a third, otherwise unused, parameter for this purpose, but that seems pretty kludged.
2. Derive `BaseCache` from `Generic[]`.  That also seems heavy handed, just for type checking.

Using a simple `Union[str, bytes]` will work as a return type for all existing cases. This could be done with a type alias at the module level. Trying to add the alias within `BaseCache` and its subclasses requires the `TypeAlias` annotation, which is only available for python >= 3.10.

I'll see whether this approach plays well with mypy.

Reply:
Yes, I meant with a `Generic`, the functions for `_get()` etc. would all need to use the same TypeVar for the key parameter. This should then type check correctly when you pass the key to memcached which expects bytes (MemcacheBackend would subclass `BaseCache[bytes]`).